File: src/main/java/at/hannibal2/skyhanni/utils/jsonobjects/CrimsonIsleReputationJson.java
package at.hannibal2.skyhanni.utils.jsonobjects;

import com.google.gson.annotations.Expose;

import java.util.List;
import java.util.Map;

public class CrimsonIsleReputationJson {
    @Expose
    public Map<String, ReputationQuest> FISHING;

    @Expose
    public Map<String, ReputationQuest> RESCUE;

    @Expose
    public Map<String, ReputationQuest> FETCH;

    @Expose
    public Map<String, ReputationQuest> DOJO;

    @Expose
    public Map<String, ReputationQuest> MINIBOSS;

    @Expose
    public Map<String, ReputationQuest> KUUDRA;

    public static class ReputationQuest {
        @Expose
        public String item;

        @Expose
        public List<Double> location;
    }
}

File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/repo/CrimsonIsleReputationJson.kt
package at.hannibal2.skyhanni.data.jsonobjects.repo

import at.hannibal2.skyhanni.utils.NeuInternalName
import com.google.gson.annotations.Expose

data class CrimsonIsleReputationJson(
    @Expose val FISHING: Map<String, ReputationQuest>,
    @Expose val RESCUE: Map<String, ReputationQuest>,
    @Expose val FETCH: Map<String, ReputationQuest>,
    @Expose val DOJO: Map<String, ReputationQuest>,
    @Expose val MINIBOSS: Map<String, ReputationQuest>,
    @Expose val KUUDRA: Map<String, ReputationQuest>,
)

data class ReputationQuest(
    @Expose val item: NeuInternalName,
    @Expose val location: List<Double>,
)
